gr-blocks/lib/message_strobe_impl.cc

```diff
@@ -91,7 +91,7 @@ namespace gr {
     void message_strobe_impl::run()
     {
       while(!d_finished) {
-        boost::this_thread::sleep(boost::posix_time::milliseconds(d_period_ms));
+        boost::this_thread::sleep(boost::posix_time::milliseconds(static_cast<long>(d_period_ms)));
```

Comment:
This highlights a mistake in message_strobe_impl!

d_period_ms is a `float`, but really, only integer milliseconds work.

So, that's a separate bug. I'm for fixing this like you propose here on both master and maint-3.7, and later change the constructor signature on master to accept `long`s instead of float. Much saner.